pytests/src/lib.rs

```diff
@@ -18,43 +15,60 @@ pub mod sequence;
-    m.add_wrapped(wrap_pymodule!(objstore::objstore))?;
-    m.add_wrapped(wrap_pymodule!(othermod::othermod))?;
-    m.add_wrapped(wrap_pymodule!(path::path))?;
-    m.add_wrapped(wrap_pymodule!(pyclasses::pyclasses))?;
-    m.add_wrapped(wrap_pymodule!(pyfunctions::pyfunctions))?;
-    m.add_wrapped(wrap_pymodule!(sequence::sequence))?;
-    m.add_wrapped(wrap_pymodule!(subclassing::subclassing))?;
-
-    // Inserting to sys.modules allows importing submodules nicely from Python
-    // e.g. import pyo3_pytests.buf_and_str as bas
-
-    let sys = PyModule::import(py, "sys")?;
-    let sys_modules: &PyDict = sys.getattr("modules")?.downcast()?;
-    sys_modules.set_item("pyo3_pytests.awaitable", m.getattr("awaitable")?)?;
-    sys_modules.set_item("pyo3_pytests.buf_and_str", m.getattr("buf_and_str")?)?;
-    sys_modules.set_item("pyo3_pytests.comparisons", m.getattr("comparisons")?)?;
-    sys_modules.set_item("pyo3_pytests.datetime", m.getattr("datetime")?)?;
-    sys_modules.set_item("pyo3_pytests.dict_iter", m.getattr("dict_iter")?)?;
-    sys_modules.set_item("pyo3_pytests.enums", m.getattr("enums")?)?;
-    sys_modules.set_item("pyo3_pytests.misc", m.getattr("misc")?)?;
-    sys_modules.set_item("pyo3_pytests.objstore", m.getattr("objstore")?)?;
-    sys_modules.set_item("pyo3_pytests.othermod", m.getattr("othermod")?)?;
-    sys_modules.set_item("pyo3_pytests.path", m.getattr("path")?)?;
-    sys_modules.set_item("pyo3_pytests.pyclasses", m.getattr("pyclasses")?)?;
-    sys_modules.set_item("pyo3_pytests.pyfunctions", m.getattr("pyfunctions")?)?;
-    sys_modules.set_item("pyo3_pytests.sequence", m.getattr("sequence")?)?;
-    sys_modules.set_item("pyo3_pytests.subclassing", m.getattr("subclassing")?)?;
+    #[pyo3]
+    use datetime::datetime;
+    #[pyo3]
+    use dict_iter::dict_iter;
+    #[pyo3]
+    use enums::enums;
+    #[pyo3]
+    use misc::misc;
+    #[pyo3]
+    use objstore::objstore;
+    #[pyo3]
+    use othermod::othermod;
+    #[pyo3]
+    use path::path;
+    #[pyo3]
+    use pyclasses::pyclasses;
+    #[pyo3]
+    use pyfunctions::pyfunctions;
+    use pyo3::types::PyDict;
+    #[pyo3]
+    use sequence::sequence;
+    #[pyo3]
+    use subclassing::subclassing;
 
-    Ok(())
+    #[pymodule_init]
+    fn init(m: &PyModule) -> PyResult<()> {
+        // Inserting to sys.modules allows importing submodules nicely from Python
+        // e.g. import pyo3_pytests.buf_and_str as bas
+        let sys = PyModule::import(m.py(), "sys")?;
+        let sys_modules: &PyDict = sys.getattr("modules")?.downcast()?;
+        sys_modules.set_item("pyo3_pytests.awaitable", m.getattr("awaitable")?)?;
+        sys_modules.set_item("pyo3_pytests.buf_and_str", m.getattr("buf_and_str")?)?;
+        sys_modules.set_item("pyo3_pytests.comparisons", m.getattr("comparisons")?)?;
+        sys_modules.set_item("pyo3_pytests.datetime", m.getattr("datetime")?)?;
+        sys_modules.set_item("pyo3_pytests.dict_iter", m.getattr("dict_iter")?)?;
```

Comment:
Maybe for a follow-up, it might be nice if this sys.modules manipulation was done automatically? (If we think it's a good idea to do that??)

Reply:
Yes! Would be great! We need to do it a bit carefully to avoid registering the module `foo.bar.baz` as also `bar.baz`. This won't be a line one change so a follow up is probably better